File: python/credit_check.py
def credit_check(string):
    #from last to first, double every other digit
    #sum of digits - on the sum of every other dig
    
    #take sum of all digs
    # modulo of 10 = 0 is true
    
    list_str = list(string)
    
    newlist = []
    summed_dig = []
    int_list =[]
    
    for i in range(0, len(list_str)):
        if i % 2 == 0:
            newlist.append(int(list_str[i]) * 2)
        else:
            newlist.append(list_str[i])
    # print(newlist, 'this is newlist')
    for j in newlist:
        if isinstance(j, int):
            if j >= 10:
                new_j = list(str(j))
                sum_dig = int(new_j[0]) + int(new_j[1])
                summed_dig.append(sum_dig)
            else:
                summed_dig.append(j)
        else:
            summed_dig.append(j)
    # print(summed_dig, 'this is summed_dig')
    for x in summed_dig:
        numbers = int(x)
        int_list.append(numbers)
    
    
    if sum(int_list) % 10 == 0:
        return 'The number is valid!'
    else:
        return 'The number is invalid!'
    
    
        


# print(credit_check('5541808923795240'))
# print(credit_check('5541801923795240'))

File: js/creditCheck.js
exports.creditCheck = function(string) {
// creditCheck = function(string) {



    let myList = string.split('')
    // console.log(myList);
    let newList = [];
    let summedDig = []
    let intList = [];

    for (let i = myList.length -1;i>=0;  i--){

        if (i % 2 ==  0) {
            newList.push(parseInt(myList[i]) * 2)
        }
        else {
            newList.push(myList[i])
        }
        
    }
    // console.log(newList)

    for (let j = 0; j < newList.length; j++) {
        if (Number.isInteger(j)) {
            if (j >= 10) {
                let newJ = String(newList[j]).split("");
                let sumDig = parseInt(newJ[0]) + parseInt(newJ[1])
                summedDig.push(sumDig)

            }
            else {
                summedDig.push(j)
            }
        }
        else {
            summedDig.push(j);

        }
        // console.log(summedDig, 'this is summedDig')
    }

    for (let x = 0; x < summedDig.length; x ++){
        let numbersNow = parseInt(summedDig[x]);
        intList.push(numbersNow)
    }

    let theSum = intList.reduce((acc,curr) => acc + curr, 0);

    if (theSum % 10 === 0) {
        return 'The number is valid!'
    }
    else {
        return 'The number is invalid!'
    }







}


// console.log(creditCheck('5541808923795240'))